### ISSUE TITLE
Catch commit conflit in parquet branch

### ISSUE THREAD

**Issue by severo** [huggingface/datasets-server#1163]
See https://huggingface.co/datasets/cais/mmlu/discussions/9.

We have to catch the exception (412) here: https://github.com/huggingface/datasets-server/blob/main/services/worker/src/worker/job_runners/config/parquet_and_info.py#L904-L911

and raise a retriable exception so that the error is only temporary.

**Comment by severo** [comment]
We have to fix this

**Comment by severo** [comment]
Note that the lock should make this error disappear. See https://github.com/huggingface/datasets-server/pull/1355 and https://github.com/huggingface/datasets-server/pull/1357.

**Comment by severo** [comment]
But we have another one:

```
Give up after 6 attempts with <class 'huggingface_hub.utils._errors.HfHubHTTPError'>

Error code:   UnexpectedError
```

which is due to timeout after waiting for too long for the lock to be released.

I think we should 1. catch this better to make the error retryable, 2. increase the duration

**Comment by severo** [comment]
cc @lhoestq

**Comment by lhoestq** [comment]
hmm actually this happens when the dataset is locked and the commits are being done.

But it seems to fail because the precondition "parent_commit" fails, though it is defined as
```python
parent_commit=parent_commit if not commit_infos else commit_infos[-1].oid
```

which sounds correct to me

**Comment by lhoestq** [comment]
Maybe it's an issue with consistency - we could add a write/read concern for the lock requests to try to fix that

**Comment by severo** [comment]
> But it seems to fail because the precondition "parent_commit" fails, though it is defined as

Indeed, it's a 412 error code, that seem to occur when we set parent_commit, and a commit has happened since
https://github.com/huggingface/huggingface_hub/blob/92028da882e47d61703e8c9144028e1ecadab415/tests/test_hf_api.py#L732

**Comment by severo** [comment]
On https://huggingface.co/datasets/gsarti/flores_101/viewer, it seems to occur for half of the configs, which is a lot!

<img width="754" alt="Capture d’écran 2023-06-20 à 13 55 12" src="https://github.com/huggingface/datasets-server/assets/1676121/134af3d5-1b03-4cef-b5a0-7861deb431e1">

**Comment by severo** [comment]
> Maybe it's an issue with consistency - we could add a write/read concern for the lock requests to try to fix that

it could be indeed

**Comment by severo** [comment]
> `parent_commit=parent_commit if not commit_infos else commit_infos[-1].oid`
> which sounds correct to me.

the best way to be sure is to add a test that checks this specifically. Do you want me to add one?

**Comment by lhoestq** [comment]
sure go ahead, thanks !

**Comment by severo** [comment]
Opened a dedicated issue for that: https://github.com/huggingface/datasets-server/issues/1396

**Comment by severo** [comment]
fixed by #1393
